app/assets/javascripts/Components/Result/autosave_text_form.jsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import { render } from "react-dom";
+
+function debounce(func, wait, immediate) {
+  var timeout;
+
+  return function executedFunction() {
+    var context = this;
+    var args = arguments;
+
+    var later = function () {
+      timeout = null;
+      if (!immediate) func.apply(context, args);
+    };
+
+    var callNow = immediate && !timeout;
+
+    clearTimeout(timeout);
+
+    timeout = setTimeout(later, wait);
+
+    if (callNow) func.apply(context, args);
+  };
+}
+
+const SaveMessage = ({ unSaved }) => (
+  <div className="autosave-text">
+    {unSaved ? (
+      <p> There are unsaved changes </p>
+    ) : (
+      <p> Changes have been saved </p>
+    )}
+  </div>
+);
+
+export class TextForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: props.initialValue,
+      unsavedChanges: false,
+    };
+    this.autoSaveText = this.autoSaveText.bind(this);
+  }
+
+  componentDidMount() {
+    this.updatePreview();
+    setInterval(this.autoSaveText, 3000);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.initialValue !== this.props.initialValue) {
+      this.setState({ value: this.props.initialValue });
+    } else if (prevState.initialValue !== this.state.value) {
+      this.updatePreview();
+    }
+  }
+
+  handlePersist = debounce(
+    () => {
+      this.props.persistChanges(this.state.value).then(() => {
+        this.setState({ unsavedChanges: false });
+      });
+    },
+    1500,
```

Comment:
I'm wondering what the relationship is between this number and the 3000. You should pull these out into constants and include a comment explaining their purpose.

Reply:
This was my bad. I forgot to push the commit that removed the `setInterval`. I'll still make this a constant